grpc-server/workflow.go

```diff
@@ -279,3 +273,17 @@ func (s *server) ShowWorkflowEvents(req *workflow.GetRequest, stream workflow.Wo
 	metrics.CacheHits.With(labels).Inc()
 	return nil
 }
+
+func getWorkflowState(db db.Database, ctx context.Context, id string) workflow.State {
+	wfctx, _ := db.GetWorkflowContexts(ctx, id)
+	if wfctx.CurrentActionState != workflow.State_STATE_SUCCESS {
+		return wfctx.CurrentActionState
+	} else {
+		wfacts, _ := db.GetWorkflowActions(ctx, id)
```

Comment:
doesnt `GetWorkflowContexts` already give `total_number_of_actions`? you could use that number and `current_action_index` to determine last action. It would save a database call.

Reply:
Yeah, that is a good observation. Actually I did it beacuse the function `isLastAction()` was implemented in that way and asks for action lists as parameter. But yeah it is not required since we have total number of actions in `wfcontext` itself. I will change that. Thanks